python/lsst/ts/standardscripts/base_take_image.py

```diff
@@ -153,6 +195,30 @@ def set_metadata(self, metadata):
             * nimages
         )
 
+    def set_additional_nextvisit_metadata(self, metadata):
+        metadata.nimages = len(self.config.exp_times)
+        metadata.coordinateSystem = MetadataCoordSys.ICRS
+        if (self.ra is not None) and (self.dec is not None):
+            radec_icrs = ICRS(
+                Angle(self.nextVisit_metadata["ra"], unit=astropy.units.hourangle),
+                Angle(self.nextVisit_metadata["dec"], unit=astropy.units.deg),
+            )
+            metadata.position = [radec_icrs.ra.deg, radec_icrs.dec.deg]
+        if self.rot_sky is not None:
+            metadata.rotationSystem = MetadataRotSys.SKY
+            metadata.cameraAngle = self.nextVisit_metadata["rot_sky"]
+            metadata.instrument = self.get_instrument_name()
+            if self.program is not None:
+                metadata.survey = self.config.program
+            instrument_configuration = self.get_instrument_configuration()
```

Comment:
Maybe instead of doing this here, that tests for the `grating` thing, make a new method that just returns the instrument configuration label. Maybe something like `get_instrument_configuration_label`? You could actually have a default implementation that just returns`instrument_configuration["filter"]` and then overwrite it in `LATISS` to return `f"{filt}~{grating}"`.